Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,190 @@
+    {beattySequence r k | k}ᶜ = {beattySequence' s k | k} := by
+  ext j
+  by_cases h₁ : j ∈ {beattySequence r k | k}
+  · by_cases h₂ : j ∈ {beattySequence' s k | k}
+    · exact (Set.not_disjoint_iff.2 ⟨j, h₁, h₂⟩ (Beatty.no_collision hrs)).elim
+    · simp only [Set.mem_compl_iff, h₁, h₂]
+  · by_cases h₂ : j ∈ {beattySequence' s k | k}
+    · simp only [Set.mem_compl_iff, h₁, h₂]
+    · have ⟨k, h₁₁, h₁₂⟩ := (Beatty.hit_or_miss hrs.pos).resolve_left h₁
+      have ⟨m, h₂₁, h₂₂⟩ := (Beatty.hit_or_miss' hrs.symm.pos).resolve_left h₂
+      exact (Beatty.no_anticollision hrs ⟨j, k, m, h₁₁, h₁₂, h₂₁, h₂₂⟩).elim
+
+theorem compl_beattySequence' {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence' r k | k}ᶜ = {beattySequence s k | k} := by
+  rw [← compl_beattySequence hrs.symm, compl_compl]
+
+/-- Generalization of Rayleigh's theorem on Beatty sequences. Let `r` be a real number greater
+than 1, and `1/r + 1/s = 1`. Then `B⁺_r` and `B⁺'_s` partition the positive integers. -/
+theorem beattySequence_symmDiff_beattySequence'_pos {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence r k | k > 0} ∆ {beattySequence' s k | k > 0} = {n | 0 < n} := by
+  apply Set.eq_of_subset_of_subset
+  · rintro j (⟨⟨k, hk, hjk⟩, -⟩ | ⟨⟨k, hk, hjk⟩, -⟩)
+    · rw [Set.mem_setOf_eq, ← hjk, beattySequence, Int.floor_pos]
+      exact one_le_mul_of_one_le_of_one_le (by norm_cast) hrs.one_lt.le
+    · rw [Set.mem_setOf_eq, ← hjk, beattySequence', ← Int.ceil_sub_one, Int.ceil_pos, sub_pos]
+      exact one_lt_mul_of_le_of_lt (by norm_cast) hrs.symm.one_lt
+  intro j (hj : 0 < j)
+  have hb₁ : ∀ s ≥ 0, j ∈ {beattySequence s k | k > 0} ↔ j ∈ {beattySequence s k | k} := by
+    intro _ hs
+    refine ⟨fun ⟨k, _, hk⟩ ↦ ⟨k, hk⟩, fun ⟨k, hk⟩ ↦ ⟨k, ?_, hk⟩⟩
+    rw [← hk, beattySequence, Int.floor_pos] at hj
+    have := pos_of_mul_pos_left (lt_of_lt_of_le zero_lt_one hj) hs
+    rwa [Int.cast_pos] at this
+  have hb₂ : ∀ s ≥ 0, j ∈ {beattySequence' s k | k > 0} ↔ j ∈ {beattySequence' s k | k} := by
+    intro _ hs
+    refine ⟨fun ⟨k, _, hk⟩ ↦ ⟨k, hk⟩, fun ⟨k, hk⟩ ↦ ⟨k, ?_, hk⟩⟩
+    rw [← hk, beattySequence', lt_sub_iff_add_lt, zero_add] at hj
+    have hj := Int.ceil_pos.1 (lt_trans zero_lt_one hj)
+    have := pos_of_mul_pos_left hj hs
+    rwa [Int.cast_pos] at this
+  rw [Set.mem_symmDiff, hb₁ _ hrs.nonneg, hb₂ _ hrs.symm.nonneg, ← compl_beattySequence hrs,
+    Set.not_mem_compl_iff, Set.mem_compl_iff, and_self, and_self]
+  exact or_not
+
+theorem beattySequence'_symmDiff_beattySequence_pos {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence' r k | k > 0} ∆ {beattySequence s k | k > 0} = {n | 0 < n} := by
+  rw [symmDiff_comm, beattySequence_symmDiff_beattySequence'_pos hrs.symm]
+
+/-- Let `r` be an irrational number. Then `B⁺_r` and `B⁺'_r` are equal. -/
+theorem Irrational.beattySequence'_pos_eq {r : ℝ} (hr : Irrational r) :
+    {beattySequence' r k | k > 0} = {beattySequence r k | k > 0} := by
```

Comment:
You should prove `beattySequence' r = beattySequence r` instead.

Reply:
I'm not sure how to express this. I'm having trouble with k=0 because `beattySeq' r 0 = -1` and `beattySeq r 0 = 0`.

Reply:
Ah of course. Then it's probably fine like that.